Mathlib/Tactic/ToAdditive/Frontend.lean

```diff
@@ -781,7 +798,9 @@ partial def transformDeclAux
       trace[to_additive_detail] "Already visited {tgt} as translation of {src}."
     return
   let srcDecl ← getConstInfo src
-  -- we first transform all auxiliary declarations generated when elaborating `pre`
+  -- we first unfold all auxlemmas, since they are not always able to be additivized on their own
+  let srcDecl ← MetaM.run' do declUnfoldAuxLemmas srcDecl
+  -- we then transform all auxiliary declarations generated when elaborating `pre`
   for n in findAuxDecls srcDecl.type pre do
     transformDeclAux cfg pre tgt_pre n
```

Comment:
Are these lines (and the few below this) still necessary?
It would be nice if you could specify in a comment somewhere which aux-decls we inline, and which ones we transform (and why). We call both of them "aux decls"/"aux lemmas", which is not very helpful.

Reply:
I've tried making the comments clearer. We're unfolding the "`proof_nnn` aux lemmas"; there are still aux decls that aren't unfolded.

I've also added a TODO to consider trying to additivize these aux lemmas and resort to unfolding them on failure.

I'm not sure exactly what kind of test to write. Any ideas?

Reply:
Maybe a test that tests whether `abstractNestedProofs`, i.e. testing for the existence of a nested abstracted proof in the body of a definition.

Maybe a test where `simp` generates an aux-lemma for the multiplicative version, where the aux-lemma cannot be additivized (by using a class with no additive counterpart, but using definitions that do have additive counterparts), but the whole declaration can be.

The last one is not that important, since I expect that Mathlib itself will have plenty of those tests.

Reply:
Could I ask you to write such test(s)? I won't be able to get to figuring out how to test this until this evening.

My motivation with this PR is getting mathlib to work on Lean v4.19.0, which is being released imminently.

Reply:
Ah, I didn't realize this was blocking the 4.19 release. Then let's merge this as is. The tests don't seem that important.